moesifwsgi/middleware.py

```diff
@@ -194,6 +195,47 @@ def background_process():
                 if self.DEBUG:
                     print('skipped')
 
+    def start_with_json(self, body):
+        return body.startswith("{") or body.startswith("[")
+
+    def transform_headers(self, headers):
+        return {k.lower(): v for k, v in headers.items()}
+
+    def base64_body(self, body):
+        return base64.standard_b64encode(body).decode(encoding="UTF-8"), "base64"
+
+    def parse_bytes_body(self, body, content_encoding, headers):
+        try:
+            if content_encoding is not None and "gzip" in content_encoding.lower() or \
+                    (headers is not None and "content-encoding" in headers and headers["content-encoding"] is not None
+                     and "gzip" in (headers["content-encoding"]).lower()):
+                parsed_body, transfer_encoding = self.base64_body(gzip.decompress(body))
```

Comment:
why are we decompressing body? wont it slow down the plugin, proly take more memory too?

Reply:
@dgilling in theory we should be able to decompress in the collector or later in the pipeline as needed?